src/components/confirm-dialog/confirm-dialog.js

```diff
@@ -0,0 +1,44 @@
+import template from "./confirm-dialog.html";
+import { MDCDialog } from "@material/dialog";
+
+export class ConfirmDialogComponent {
+  constructor(mountPoint, props) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  showDialog() {
```

Comment:
probably it would be better to pass `type` and `name` via `showDialog(type, name)` -- so please remove these props and set type and name in `showDialog(type, name)`